src/synapse/azext_synapse/_validators.py

```diff
@@ -3,19 +3,69 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
+import threading
+from knack.util import CLIError
+from msrestazure.tools import is_valid_resource_id
+from azure.cli.core.commands.parameters import get_resources_in_subscription  # pylint: disable=unused-import
+from .util import get_resource_id_by_name
 
-def example_name_or_id_validator(cmd, namespace):
-    #  pylint: disable=line-too-long
-    # Example of a storage account name or ID validator.
-    # See: https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-name-or-id-parameters
-    from azure.cli.core.commands.client_factory import get_subscription_id
-    from msrestazure.tools import is_valid_resource_id, resource_id
-    if namespace.storage_account:
-        if not is_valid_resource_id(namespace.RESOURCE):
-            namespace.storage_account = resource_id(
-                subscription=get_subscription_id(cmd.cli_ctx),
-                resource_group=namespace.resource_group_name,
-                namespace='Microsoft.Storage',
-                type='storageAccounts',
-                name=namespace.storage_account
-            )
+VALIDATION_TIME_OUT = 20
+
+
+def validate_storage_account(namespace):
+    from msrestazure.tools import parse_resource_id
+    if is_valid_resource_id(namespace.storage_account):
+        parsed_workspace = parse_resource_id(namespace.storage_account)
+        storage_name = parsed_workspace['resource_name']
+        namespace.storage_account = storage_name
+
+
+def validate_workspace(cmd, namespace):

```

Comment:
Since Azure CLI has the `id_part` mechanism for resource id workflow. Is this complicated logic really neccessary?

Reply:
> Since Azure CLI has the `id_part` mechanism for resource id workflow. Is this complicated logic really neccessary?

id_part is related with support --ids. And create command don't support --ids. I use this validate to support providing workspace's id when create a spark pool or sql pool. just like:
az synpase spark pool create --workspace /sub/xx/xxxx/worksapcename

Reply:
Let's discuss this offline. Maybe we should take a look at child resource creation behavior of some existing services. Anyway this code looks not quite elegant.. I wrote them by myself before but I don't suggest to use it.. Sorry for the misleading...

Reply:
> Let's discuss this offline. Maybe we should take a look at child resource creation behavior of some existing services. Anyway this code looks not quite elegant.. I wrote them by myself before but I don't suggest to use it.. Sorry for the misleading...

removed validate_workspace